sr_data_visualization/src/sr_data_visualization/data_plot.py

```diff
@@ -91,17 +91,24 @@ def timerEvent(self):
 
         self.replot()
 
-    def plot_data(self, plot):
+    def plot_data(self, plot, side, new_sub=True):
+        self._topic_name = self._topic_name.replace("lh", side).replace("rh", side)
+
         if plot:
+            if self._subscriber:
```

Comment:
wouldn't be better to have just at the beginning a call that if subscriber is a thing we unresigter and clear. In this way we would avoid having code repetition. Because now if plot we unregister and clear, if we don't plot we also unregister and clear.

Reply:
That's right, I changed that.